### PR TITLE
Collect more rapid task info

### PR DESCRIPTION
**Pull request by jontje**
As per title

### CODE REVIEW

File: include/abb_librws/rws_interface.h [hunk 1]
```diff
@@ -131,11 +143,18 @@ class RWSInterface
...
      * \param name for the name of the task.
      * \param is_motion_task indicating if the task is a motion task or not.
+     * \param is_active indicating if the task is active or not.
+     * \param execution_state indicating the task's current execution state.
      */
-    RAPIDTaskInfo(const std::string& name, bool is_motion_task)
+    RAPIDTaskInfo(const std::string& name,
+                  const bool is_motion_task,
+                  const bool is_active,
+                  const RAPIDTaskExecutionState execution_state)
```
Comment (gavanderhoorn) [inline include/abb_librws/rws_interface.h:152]: Is this a breaking change (ie: users could already be using this)?

File: src/rws_interface.cpp [hunk 2]
```diff
@@ -264,8 +264,28 @@ std::vector<RWSInterface::RAPIDTaskInfo> RWSInterface::getRAPIDTasks()
   {
     std::string name = xmlFindTextContent(node_list.at(i), XMLAttributes::CLASS_NAME);
     bool is_motion_task = xmlFindTextContent(node_list.at(i), XMLAttributes::CLASS_MOTIONTASK) == RAPID::RAPID_TRUE;
+    bool is_active = xmlFindTextContent(node_list.at(i), XMLAttributes::CLASS_ACTIVE) == "On";
+    std::string temp = xmlFindTextContent(node_list.at(i), XMLAttributes::CLASS_EXCSTATE);
 
-    result.push_back(RAPIDTaskInfo(name, is_motion_task));
+    RAPIDTaskExecutionState execution_state = UNKNOWN;
```
Comment (gavanderhoorn) [inline src/rws_interface.cpp:270]:
```suggestion
    // assume task state is unknown, update based on contents of 'temp'
    RAPIDTaskExecutionState execution_state = UNKNOWN;
```

File: src/rws_interface.cpp [hunk 3]
```diff
@@ -264,8 +264,28 @@ std::vector<RWSInterface::RAPIDTaskInfo> RWSInterface::getRAPIDTasks()
...
+    }
+    else if(temp == "star")
+    {
+      execution_state = STARTED;
+    }
+    else if(temp == "unin")
+    {
+      execution_state = UNINITIALIZED;
+    }
+
```
Comment (gavanderhoorn) [inline src/rws_interface.cpp:287]: Only after reading this and looking back (to line 270) did I realise "any other value" in `temp` was handled by the default being `UNKNOWN`.

**Review by gavanderhoorn — COMMENTED** [review]
Some more comments.

File: include/abb_librws/rws_interface.h (hunk 1 above)
Comment (jontje) [reply include/abb_librws/rws_interface.h:152]: No, I think that is very unlikely.

The construction of `RAPIDTaskInfo` objects should only occur inside the `getRAPIDTasks(...)` method [here](https://github.com/jontje/abb_librws/blob/1f8478e5ff115129f0454208212263d33f709ada/src/rws_interface.cpp#L288).

**Review by gavanderhoorn — DISMISSED** [review]

**Comment by jontje** [comment]
@gavanderhoorn, can you reapprove the PR? I have just synced it with the base branch.

**Review by gavanderhoorn — APPROVED** [review]

**Comment by jontje** [comment]
Thanks for the review @gavanderhoorn 👍

**Pull request merged**